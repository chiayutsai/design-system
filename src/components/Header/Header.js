/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import useStyles from 'isomorphic-style-loader/useStyles'
import Link from '../Link'
import Navigation from '../Navigation'
import logoUrl from './assets/logo-small.png'
import logoUrl2x from './assets/logo-small@2x.png'
import styles from './Header.scss'

const Header = () => {
  useStyles(styles)
  return (
    <div className={styles.root}>
      <div className={styles.container}>
        <Navigation />
        <Link className={styles.brand} to="/">
          <img
            src={logoUrl}
            srcSet={`${logoUrl2x} 2x`}
            width="38"
            height="38"
            alt="React"
          />
          <span className={styles['brand-txt']}>Your Company</span>
        </Link>
        <div className={styles.banner}>
          <h1 className={styles['banner-title']}>React</h1>
          <p className={styles['banner-desc']}>Complex web apps made easy</p>
        </div>
      </div>
    </div>
  )
}

export default Header
