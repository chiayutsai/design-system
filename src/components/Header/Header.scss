/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

@import '~components/variables';

:root {
  --brand-color: #61dafb;
}

.root {
  background: #373277;
  color: #fff;
}

.container {
  margin: 0 auto;
  padding: 20px 0;
  max-width: var(--max-content-width);
}

.brand {
  color: var(--brand-color);
  text-decoration: none;
  font-size: 1.75em; /* ~28px */
}

.brand-txt {
  margin-left: 10px;
}

.banner {
  text-align: center;
}

.banner-title {
  margin: 0;
  padding: 10px;
  font-weight: normal;
  font-size: 4em;
  line-height: 1em;
}

.banner-desc {
  padding: 0;
  color: rgba(255, 255, 255, 0.5);
  font-size: 1.25em;
  margin: 0;
}
