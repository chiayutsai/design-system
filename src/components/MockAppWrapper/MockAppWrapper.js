import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
