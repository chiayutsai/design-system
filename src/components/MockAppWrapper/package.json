{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
