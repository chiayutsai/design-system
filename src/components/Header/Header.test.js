import renderer from 'react-test-renderer'
import MockAppWrapper from 'components/MockAppWrapper'
import Header from './Header'

describe('Header', () => {
  test('render correctly', () => {
    const wrapper = renderer
      .create(
        <MockAppWrapper>
          <Header />
        </MockAppWrapper>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
