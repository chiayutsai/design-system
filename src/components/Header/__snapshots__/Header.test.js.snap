// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <div
      className="root"
      role="navigation"
    >
      <a
        className="link"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <a
        className="link"
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
      <span
        className="spacer"
      >
         | 
      </span>
      <a
        className="link"
        href="/login"
        onClick={[Function]}
      >
        Log in
      </a>
      <span
        className="spacer"
      >
        or
      </span>
      <a
        className="link highlight"
        href="/register"
        onClick={[Function]}
      >
        Sign up
      </a>
    </div>
    <a
      className="brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="React"
        height="38"
        src="logo-small.png"
        srcSet="logo-small@2x.png 2x"
        width="38"
      />
      <span
        className="brand-txt"
      >
        Your Company
      </span>
    </a>
    <div
      className="banner"
    >
      <h1
        className="banner-title"
      >
        React
      </h1>
      <p
        className="banner-desc"
      >
        Complex web apps made easy
      </p>
    </div>
  </div>
</div>
`;
